### PR TITLE
feat: implement disabling logging api

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

**Review by dmitry-fa — CHANGES_REQUESTED** [review]
I'm not sure that `disableLogging()` is a good method for a builder class.
Just supporting `setLogging(null)` will be enough, I think

**Comment by athakor** [comment]
@dmitry-fa According to the given [api doc](https://cloud.google.com/storage/docs/access-logs#disabling), this method seems more appropriate than to set null.

Same way implemented in python as well see [disable_logging](https://googleapis.dev/python/storage/latest/_modules/google/cloud/storage/bucket.html#Bucket.disable_logging)

**Comment by dmitry-fa** [comment]
@athakor, 
there is a well-known software design pattern called 'Builder'. This pattern is widely used in Google libraries. The typical usage is:    
`updated =  obj.toBuilder().setX(x).setY(y).setZ(z).build()`
 The problem of the current implementation that `setLogging(null)` is not handled correctly. 
Your change fixes that problem. That's great. 
But the fix brings also inconsistency with a generic approach, which is not necessary. 

If you don't believe me, you can submit a public PR. But I will raise the same question there.

**Comment by dmitry-fa** [comment]
> 
> @dmitry-fa According to the given [api doc](https://cloud.google.com/storage/docs/access-logs#disabling), this method seems more appropriate than to set null.
> 

Quoting an [example ](https://cloud.google.com/storage/docs/access-logs#json-api_2) from the doc you mentioned:

```
PATCH /example-bucket?logging HTTP/1.1
Host: storage.googleapis.com
{
 "logging": null
}
```
Disabling logging is implemented as setting the logging property to null

**Comment by athakor** [comment]
> > @dmitry-fa According to the given [api doc](https://cloud.google.com/storage/docs/access-logs#disabling), this method seems more appropriate than to set null.
> 
> Quoting an [example ](https://cloud.google.com/storage/docs/access-logs#json-api_2) from the doc you mentioned:
> 
> ```
> PATCH /example-bucket?logging HTTP/1.1
> Host: storage.googleapis.com
> {
>  "logging": null
> }
> ```
> 
> Disabling logging is implemented as setting the logging property to null

okay thanks for the feedback will did the change as per  you.

**Comment by athakor** [comment]
@dmitry-fa PTAL

File: google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java [hunk 1]
```diff
@@ -3322,4 +3322,37 @@ public void testDeleteLifecycleRules() throws ExecutionException, InterruptedExc
...
+
+      assertEquals(logsBucket, bucket.getLogging().getLogBucket());
+      assertEquals("test-logs", bucket.getLogging().getLogObjectPrefix());
+      Bucket updatedBucket = bucket.toBuilder().setLogging(null).build().update();
+      assertThat(updatedBucket.getLogging()).isNull();
+    } finally {
+      RemoteStorageHelper.forceDelete(storage, logsBucket, 5, TimeUnit.SECONDS);
+      RemoteStorageHelper.forceDelete(storage, loggingBucket, 5, TimeUnit.SECONDS);
+    }
+  }
```
Comment (dmitry-fa) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: You duplicated the code of an existed tests, that is not good. 
Could you extend the existing test to check that set loggin to null have an effect, please

File: google-cloud-storage/src/test/java/com/google/cloud/storage/BucketTest.java [hunk 2]
```diff
@@ -863,4 +863,51 @@ public void testDeleteLifecycleRules() {
...
+            .setStorageClass(STORAGE_CLASS)
+            .setVersioningEnabled(VERSIONING_ENABLED)
+            .setLabels(BUCKET_LABELS)
+            .setRequesterPays(REQUESTER_PAYS)
+            .setDefaultKmsKeyName(DEFAULT_KMS_KEY_NAME)
+            .setDefaultEventBasedHold(DEFAULT_EVENT_BASED_HOLD)
+            .setRetentionEffectiveTime(RETENTION_EFFECTIVE_TIME)
+            .setRetentionPeriod(RETENTION_PERIOD)
+            .setRetentionPolicyIsLocked(RETENTION_POLICY_IS_LOCKED)
+            .build();
```
Comment (dmitry-fa) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/BucketTest.java]: Nit: do you need all these settings here? Would it make sense to minimize invocation of setters on  the builder, just to make the code more compact.

**Review by dmitry-fa — CHANGES_REQUESTED** [review]
Overall looks good, please fix to avoid duplication in the ITStorage test

**Comment by athakor** [comment]
@dmitry-fa both the comments have been addressed PTAL

**Pull request closed without merging**